src/consistent_request.rs

```diff
@@ -0,0 +1,226 @@
+use std::{
+    sync::{Arc, atomic::AtomicBool},
+    time::Duration,
+};
+
+use eyre::{bail, eyre};
+use http::{request, response};
+use jsonrpsee::{
+    core::{BoxError, http_helpers},
+    http_client::{HttpBody, HttpRequest, HttpResponse},
+};
+use parking_lot::Mutex;
+use tokio::{sync::watch, task::JoinHandle};
+use tracing::{error, info, warn};
+
+use crate::{ExecutionMode, Health, HttpClient, Probes};
+
+/// A request manager that ensures requests are sent consistently
+/// across both the builder and l2 client.
+#[derive(Clone, Debug)]
+pub struct ConsistentRequest {
+    method: String,
+    l2_client: HttpClient,
+    builder_client: HttpClient,
+    has_disabled_execution_mode: Arc<AtomicBool>,
+    req_tx: watch::Sender<Option<(request::Parts, Vec<u8>)>>,
+    res_rx: watch::Receiver<Option<Result<(response::Parts, Vec<u8>), BoxError>>>,
+    probes: Arc<Probes>,
+    execution_mode: Arc<Mutex<ExecutionMode>>,
+}
+
+impl ConsistentRequest {
+    /// Creates a new `ConsistentRequest` instance.
+    pub fn new(
+        method: String,
+        l2_client: HttpClient,
+        builder_client: HttpClient,
+        probes: Arc<Probes>,
+        execution_mode: Arc<Mutex<ExecutionMode>>,
+    ) -> Self {
+        let (req_tx, mut req_rx) = watch::channel(None);
+        let (res_tx, mut res_rx) = watch::channel(None);
+        req_rx.mark_unchanged();
+        res_rx.mark_unchanged();
+
+        let has_disabled_execution_mode = Arc::new(AtomicBool::new(false));
```

Comment:
why is this initially set to false? shouldn't it be derived from the execution mode?